elastica/boundary_conditions.py

```diff
@@ -30,44 +33,50 @@ class ConstraintBase(ABC):
 
     """
 
-    _system: SystemType
-    _constrained_position_idx: np.ndarray
-    _constrained_director_idx: np.ndarray
+    _system: S
+    _constrained_position_idx: NDArray[np.integer]
+    _constrained_director_idx: NDArray[np.integer]
 
-    def __init__(self, *args, **kwargs):
+    def __init__(
+        self,
+        *args: Any,
+        constrained_position_idx: ConstrainingIndex = (),
+        constrained_director_idx: ConstrainingIndex = (),
+        **kwargs: Any,
+    ) -> None:
         """Initialize boundary condition"""
         try:
             self._system = kwargs["_system"]
             self._constrained_position_idx = np.array(
-                kwargs.get("constrained_position_idx", []), dtype=int
+                constrained_position_idx, dtype=np.int_
```

Comment:
Numpy said: ` DeprecationWarning: Converting `np.integer` or `np.signedinteger` to a dtype is deprecated.`

Although, I think it is probably a good idea to remove ambiguity. We should stick with either np.int32 or np.int64 to be clear.

Reply:
Probably same for `np.floating` -> `np.float64`

Reply:
I removed ambiguity comes with numpy data type. Lets just stick with `int32` and `float64`.

[e3ac13e](https://github.com/GazzolaLab/PyElastica/pull/367/commits/e3ac13eeb889de40c82ae7675c9bdc300ccbdc8e)

Reply:
lets do np.int64 instead of 32

Reply:
Why?

Reply:
Int32 is faster and less memory, and it is still native for most of the numpy. Int_ switch int32 and int64 depends on the hardware, but we should remove ambiguity since we do numerics. we practically dont need int64. Every hardware, including HPCs and linalg packages are more tuned for int32. We should keep int32